aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -126,6 +128,14 @@ TensorIterator make_value_selection_intersection_iter(
     return values.as_strided(values_sizes, values_strides);
   };
 
+  const auto match_mask = [&match_mask_opt, &lhs_select_idx]() -> Tensor {
```

Comment:
Nit (@pearu pointed it to me enough time so that I remembered it :-)):
```C++
const auto match_mask = match_mask_opt.value_or(at::ones_like(lhs_select_idx));
```

Reply:
Hm, thanks! This is something new :)

Reply:
c10::optional is a forward port of [std::optional](https://en.cppreference.com/w/cpp/utility/optional) so it should follow that API. Now that we're on C++17 for PT it might turn into a typedef etc.

Reply:
No, because of the Christian's comment.

Reply:
OK, that makes sense. I would also be worried if `at::ones_like(lhs_select_idx)` is unnecessarily evaluated in when the match_mask value is specified.